src/main/scala/uk/gov/hmrc/ct/computations/CP87Input.scala

```diff
@@ -16,12 +16,37 @@
 
 package uk.gov.hmrc.ct.computations
 
-import uk.gov.hmrc.ct.box.{CtBoxIdentifier, CtOptionalInteger, Input}
+import uk.gov.hmrc.ct.box._
+import uk.gov.hmrc.ct.computations.Validators.ComputationValidatableBox
+import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
 
-case class CP87Input(value: Option[Int]) extends CtBoxIdentifier(name = "First year allowance claimed") with CtOptionalInteger with Input
+case class CP87Input(value: Option[Int]) extends CtBoxIdentifier(name = "First year allowance claimed")  with CtOptionalInteger with Input with ComputationValidatableBox[ComputationsBoxRetriever] {
+  override def validate(boxRetriever: ComputationsBoxRetriever) = {
+    validateZeroOrPositiveInteger(this) ++
+      mandatoryIfCompanyIsTradingTrading(boxRetriever,"CP87Input", value) ++
+      firstYearAllowanceNotGreaterThanMaxFYA(boxRetriever)
+  }
+
+  private def firstYearAllowanceNotGreaterThanMaxFYA(retriever: ComputationsBoxRetriever): Set[CtValidation] = {
+
+    val expenditureQualifyingForFirstYearAllowanceInput = retriever.retrieveCP81Input()
+    val cpAux1 = retriever.retrieveCPAux1()
+
+    val maxFYA = expenditureQualifyingForFirstYearAllowanceInput.orZero + cpAux1
+
+    value match {
+      case Some(fyaClaimed) if fyaClaimed > maxFYA => Set()
```

Comment:
You've got the empty Set here when you want the line below to return, right? It will be returned anyway I guess.